lib/parser.ml

```diff
@@ -0,0 +1,33 @@
+open Omd
+
+let extract_links text =
+  let md = of_string text in
+  let rec loop = function
+    | [] -> []
+    | Url (href, s, _) :: tl ->
+        let lst = href :: loop s in
+        lst @ loop tl
+    | Blockquote hd :: tl
+    | Paragraph hd :: tl
+    | Emph hd :: tl
+    | Bold hd :: tl
+    | H1 hd :: tl
+    | H2 hd :: tl
+    | H3 hd :: tl
+    | H4 hd :: tl
+    | H5 hd :: tl
+    | H6 hd :: tl ->
+        let lst = loop hd in
+        lst @ loop tl
```

Comment:
This can also be simplified using the same syntax as the one used to match `Ul` and `Ol`